enclave_build/src/lib.rs

```diff
@@ -58,8 +63,15 @@ pub enum Docker2EifError {
 }
 
 impl<'a> Docker2Eif<'a> {
+    /// If the 'docker_dir' argument is Some (i.e. --docker-dir flag is used), the docker daemon will
+    /// always be used to build the image locally and store it in the docker cache.
+    ///
+    /// If the 'oci_image' argument is Some (i.e. the --image-uri flag is used) and the '--docker-dir'
+    /// flag is not used, pull and cache the image without using the docker daemon.
     pub fn new(
         docker_image: String,
+        docker_dir: Option<String>,
+        oci_image: Option<String>,
```

Comment:
Shouldn't these 3 be exclusive to each other? Perhaps use a single enum instead?

Reply:
For backwards compatibility I kept the '--docker-uri' flag required, and if the '--oci-uri' flag is used, then the argument of the first one is ignored.

Reply:
That is quite confusing IMO. The CLI shouldn't accept both --docker-uri and --oci-uri at the same time.

Reply:
It should be either --docker-uri with optional --docker-dir or --oci-uri